workspace/manager.py

```
@@ -240,12 +253,157 @@ def create_workspace(
+            dict[str, object]: {"ok": True, "error": None} when the ref is found; {"ok": False, "error": <message>} otherwise.
+        """
+        if not repo_url or not ref:
+            return {"ok": False, "error": "missing_repo_or_ref"}
+        try:
+            import subprocess
+            env = dict(**os.environ)
+            env.setdefault("GIT_TERMINAL_PROMPT", "0")
+            auth_url = repo_url
+            if token and repo_url.startswith("https://"):
+                auth_url = repo_url.replace("https://", f"https://{token}@")
+            proc = subprocess.run(["git", "ls-remote", "--heads", auth_url, ref], stdout=subprocess.PIPE, stderr=subprocess.PIPE, env=env, timeout=timeout)
```

Ruff (0.15.12)

[error] `subprocess` call: check for execution of untrusted input

(S603)